code/modules/mob/living/simple_animal/friendly/trader.dm

```diff
@@ -94,154 +181,285 @@
 		return FALSE
 	return TRUE
 
-/**
- * Tries to call sell_item on one of the user's held items, if fail gives a chat message
- *
- * Gets both items in the user's hands, and then tries to call sell_item on them, if both fail, he gives a chat message
- * Arguments:
- * * user - The mob trying to sell something
- */
-/mob/living/simple_animal/hostile/retaliate/trader/proc/try_sell(mob/user)
-	var/obj/item/activehanditem = user.get_active_held_item()
-	var/obj/item/inactivehanditem = user.get_inactive_held_item()
-	if(!sell_item(user, activehanditem) && !sell_item(user, inactivehanditem))
-		say(itemrejectphrase)
+///Talk about what items are being sold/wanted by the trader and in what quantity or lore
+/mob/living/simple_animal/hostile/retaliate/trader/proc/discuss(mob/user)
+	var/list/npc_options = list(
+		"Lore" = image(icon = 'icons/hud/radial.dmi', icon_state = "radial_lore"),
+		"Selling?" = image(icon = 'icons/hud/radial.dmi', icon_state = "radial_selling"),
+		"Buying?" = image(icon = 'icons/hud/radial.dmi', icon_state = "radial_buying")
+	)
+	var/pick = show_radial_menu(user, src, npc_options, custom_check = CALLBACK(src, .proc/check_menu, user), require_near = TRUE, tooltips = TRUE)
+	if(pick == "Lore")
+		say(return_trader_phrase(TRADER_LORE_PHRASE))
+	if(pick == "Buying?")
+		trader_buys_what(user)
+	if(pick == "Selling?")
+		trader_sells_what(user)
 
-///Makes the NPC say one picked thing from the lore list variable, can be overriden for fun stuff
-/mob/living/simple_animal/hostile/retaliate/trader/proc/deep_lore()
-	say(pick(lore))
+///Displays to the user what the trader is willing to buy and how much until a restock happens
+/mob/living/simple_animal/hostile/retaliate/trader/proc/trader_buys_what(mob/user)
+	if(!wanted_items.len)
+		to_chat(user, span_green("I'm currently buying nothing at the moment."))
```

Comment:
Can all these be replaced with say(), so the trader is physically speaking like a person, while also adding runechat to this?